.circleci/config.yml

```diff
@@ -114,6 +114,27 @@ commands:
       - brew_install:
           formulae: libtool
 
+  maybe_rebase:
+    steps:
+      - run:
+          name: Rebase on top of REBASE_TARGET if exists
+          no_output_timeout: "10m"
+          command: |
+            set -ex
+            if [ -n "$REBASE_TARGET" ] && [ -n "$CIRCLE_PULL_REQUEST" ]; then
+              git config --global user.email "circleci.ossci@gmail.com"
+              git config --global user.name "CircleCI"
+              PR_NUM=$(basename $CIRCLE_PULL_REQUEST)
+              CIRCLE_PR_BASE_BRANCH=$(curl -s https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$PR_NUM | jq -r '.base.ref')
+              echo "Rebase on top of $CIRCLE_PR_BASE_BRANCH branch before building in environment $BUILD_ENVIRONMENT"
+              if [[ "${CIRCLE_PR_BASE_BRANCH}" == "master" ]]; then
```

Comment:
Why we would like to have separate handling for master based PR?

Reply:
Hmmm so I was thinking about whether to make xla job rebase on top `viable/strict` or `origin/master` so I opened a special case here that we could pass a param in. 
I can indeed simplify the logic if we agree `origin/master` is a better option. cc: @ezyang @albanD wdyt?

Reply:
master is not broken too often but we don't want chain failure where all CI fails as soon as master break I think.

Reply:
It should be based off of the base branch that the PR is basing its changes off of.

Hardcoding `origin/master` should never be an option because that immediately breaks as soon as we do a release branch.